Macrophages/Macrophages_multiome.py

```diff
@@ -0,0 +1,746 @@
+    return imputed_df
+
+
+def iterative_imputer(df, maxiteration=10, randomstates=0):
+    from sklearn.impute import IterativeImputer
+
+    # Initialize the IterativeImputer
+    imputer = IterativeImputer(max_iter=maxiteration, random_state=randomstates)
+
+    # Create a new DataFrame with the imputed values
+    imputed_df = pd.DataFrame(imputer.fit_transform(df), columns=df.columns, index=df.index)
+    return imputed_df
+
+
+def standardscaler(df):
+    # # Initialize the StandardScaler
+    scaler = StandardScaler()
+    # Fit and transform the data according to rows
+    scaled_data = scaler.fit_transform(df)
+    scaled_df = pd.DataFrame(scaled_data, index=df.index, columns=df.columns)
+    return scaled_df
+
+
+def calculate_mean(df):
+    # Calculate mean values for each group of columns
+    mean_T1 = df.iloc[:, 0:6].mean(axis=1)
+    mean_T2 = df.iloc[:, 6:12].mean(axis=1)
+    mean_T3 = df.iloc[:, 12:18].mean(axis=1)
+    mean_T4 = df.iloc[:, 18:24].mean(axis=1)
+    # Create a new DataFrame with the calculated means
+    new_df = pd.DataFrame({
+        'T1': mean_T1,
+        'T2': mean_T2,
+        'T3': mean_T3,
+        'T4': mean_T4
+    })
+    return new_df
+
+def filter_row_missings(df, number):
+    '''Filter protein features and keep proteins with less than "number" missing values, protein features as index.'''
+    # Using isna() to count missing values and keep rows with missing values less than "number"
+    df_filtered = df[df.isna().sum(axis=1) <= number]
+    return df_filtered
+
+def filter_column_missings(df, value):
+    '''Filter samples and keep samples with more than "value" protein identifications, samples as columns.'''
+    # Using notna() to count non-missing values and keep columns with non-missing values more than "value"
+    df_filtered = df.loc[:, df.notna().sum(axis=0) >= value]
+    return df_filtered
+
+# function for oneway_ANOVA analysis
+def oneway_ANOVA_Ttest(result):
+    pval = []
+    for i in range(len(result)):
+        aa = result.iloc[i, :6].tolist()
+        bb = result.iloc[i, 6:12].tolist()
+        cc = result.iloc[i, 12:18].tolist()
+        dd = result.iloc[i, 18:24].tolist()
+        p_value_anova = f_oneway(aa, bb, cc, dd)[1]
+        p_value_ttest = stats.ttest_ind(aa, dd)[1]  # ttest between irradiation and control
+        pval.append([result.index[i], p_value_anova, p_value_ttest])
+        fl = pd.DataFrame(pval)
+        fl.columns = ['protein', 'p_value_anova', 'p_value_ttest']
+    return (fl)
+
+# 按行进行标准归一化
+def standardscaler_row(df):
+    scaler = StandardScaler()
+    proname = df.T.columns
+    k = df.T
+    dfdf_pro = scaler.fit_transform(k)
+    # 按行进行标准化后的 dataframe
+    dfpro_stdbyrow = pd.DataFrame(dfdf_pro.T, index=df.index, columns=df.columns)
+    return dfpro_stdbyrow
+
+# set four treatments 
+treatment=['Con']*6 + ['LPS']*6 + ['IL_4']*6 + ['IRD']*6
+
+def getdf(protein_name = '1/sp|B2RXS4|PLXB2_MOUSE',df = df_lfq):

```

Comment:
`df = df_lfq` relies on df_lfq being defined. This works for how you have it, but each time this file is called (e.g. to use one of the processing functions), your entire dataset gets loaded. You generally want to avoid having default variables be variables.